temporal-opentracing/src/main/java/io/temporal/opentracing/internal/OpenTracingWorkflowInboundCallsInterceptor.java

```diff
@@ -0,0 +1,68 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.opentracing.internal;
+
+import io.opentracing.Scope;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+import io.temporal.common.interceptors.WorkflowInboundCallsInterceptor;
+import io.temporal.common.interceptors.WorkflowInboundCallsInterceptorBase;
+import io.temporal.common.interceptors.WorkflowOutboundCallsInterceptor;
+import io.temporal.opentracing.OpenTracingConfig;
+import io.temporal.workflow.Workflow;
+
+public class OpenTracingWorkflowInboundCallsInterceptor
+    extends WorkflowInboundCallsInterceptorBase {
+  private final OpenTracingConfig openTracingConfig = OpenTracingConfig.newBuilder().build();
+
+  public OpenTracingWorkflowInboundCallsInterceptor(WorkflowInboundCallsInterceptor next) {
+    super(next);
+  }
+
+  @Override
+  public void init(WorkflowOutboundCallsInterceptor outboundCalls) {
+    super.init(new OpenTracingWorkflowOutboundCallsInterceptor(outboundCalls));
+  }
+
+  @Override
+  public WorkflowOutput execute(WorkflowInput input) {
+    Tracer tracer = GlobalTracer.get();
+    SpanContext rootSpanContext =
+        OpenTracingContextPropagationUtils.readSpanContextFromHeader(input.getHeader(), tracer);
+    Span workflowRunSpan =
+        OpenTracingSpanUtils.createWorkflowRunSpan(
+                tracer,
+                Workflow.currentTimeMillis(),
+                openTracingConfig.getWorkflowRunOperationNamePrefix()
+                    + "-"
+                    + Workflow.getInfo().getWorkflowType(),
```

Comment:
Why do we need to add workflow type at the end? Should we add it into tags instead?

Reply:
Just WorkflowStart or WorkflowRun as an operation name is very undescriptive and pretty meaningless.

A definition/best practice from [OpenTracing Specification](https://github.com/opentracing/specification/blob/master/specification.md):

> An operation name, a human-readable string which concisely represents the work done by the Span (for example, an RPC method name, a function name, or the name of a subtask or stage within a larger computation). The operation name should be the most general string that identifies a (statistically) interesting class of Span instances. That is, "get_user" is better than "get_user/314159".

Also from [Open Tracing Guide](https://opentracing.io/guides/golang/spans/):

> The reason for choosing general operation names is to allow the tracing systems to do aggregations. For example, Jaeger tracer has an option of emitting metrics for all the traffic going through the application. Having a unique operation name for each span would make the metrics useless. For application or library developers who wish to capture the program arguments in the traces to distinguish them, the recommended solution is to annotate spans with tags or logs. These are discussed in the section below.
> 
> Each operation name should have some semantic meaning to it and should be chosen such that its cardinality is neither too high nor too low. The cardinality of an operation name is defined as follows:
> 
> card(operation X) = total number of spans of operation X
> 
> For example, when using http.url (as operation name) the cardinality would be too high, on the other hand if http.method is chosen as the operation name then the cardinality would be too low.

I think that my current solution is most in line with these guidelines.

Reply:
It actually goes against what you've quoted above, specifically "operation name should be the most general string" as we concatenate specific workflow type to operation.

Reply:
But you just took the fragment out of the context. The full phrase is "most general string that identifies a (statistically) interesting class of Span instances". And there are more phrases around that explain what does it mean. If you open the first link, after this phrase there is a specific example with good and bad examples:

Operation Name | Guidance
-- | --
get | Too general
get_account/792 | Too specific
get_account | Good, and account_id=792 would make a nice Span tag

You offer to make an operation name EVEN MORE generic thing than this `get` example.
I'm standing on the fact that I spent a good amount of time investigating the different implementation and open tracing guidelines and my current solution is much close to the guidelines.

Reply:
This is how OpenTelemetry UIs look like:
![c6463a2b-trace-300](https://user-images.githubusercontent.com/532108/113498093-d6597b00-94d7-11eb-8b6f-dd54cfd91e2e.png)
![0_WDqZqICj-V23_1YW](https://user-images.githubusercontent.com/532108/113498095-df4a4c80-94d7-11eb-84c8-bd4355e19d77.png)
![aanuraag_Distributed-Tracing-OpenTelemetry_f1](https://user-images.githubusercontent.com/532108/113498123-1b7dad00-94d8-11eb-959f-dd979c34eb30.jpeg)
To get to tags you need to click on a specific span and open the details. How the operator will work with a sequence of indistinguishable ActivityRun spans?

Also, see OpenTracing implementation for JaxRS2:
https://github.com/opentracing-contrib/java-jaxrs/blob/master/opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/OperationNameProvider.java#L46

> Default Microprofile operation name <HTTP method>:<package name>.<Class name>.<method name>
<package name>.<Class name>.<method name> is our ActivityType/WorkflowType

They do provide an alternative strategy
```
  /**
   * Returns HTTP method as operation name
   */
  class HTTPMethodOperationName implements OperationNameProvider {
```
but it's not a default